### PR TITLE
Message - add a few methods

### PR DESCRIPTION
**Pull request by gautam-paranjape**
I added a delete method to the Message structure, currently, it doesn't work as when sending the request you need to define the channel, and we don't have the Channel structure yet, so I will update it once we create that structure. 

Don't merge the PR yet, I am still going to add a few more methods.

### CODE REVIEW

File: src/structures/Message.ts [hunk 1]
```diff
@@ -39,12 +42,24 @@ export class Message extends Base {
   public async delete(options: MessageDeleteOptions): Promise<Message> {
     return new Promise((resolve, reject) => {
       setTimeout(async () => {
-        return await this.client.api.request({
+        await this.client.api.request({
           method: "DELETE",
-          endpoint: `channels/channel/messages/${this.id}`,
+          endpoint: `channels/channelID/messages/${this.id}`,
```
Comment (PraveshKunwar) [inline src/structures/Message.ts:47]: This channelID will need to be replaced. Need to add a property called channel that will have a typeof text / dm / newschannel.  essentially

```ts
private channel: TextChannel | DMChannel | NewsChannel; 

//down at the delete method
endpoint: `channels/${this.channel.id}/messages/${this.id}`
```

File: src/structures/Message.ts [hunk 2]
```diff
@@ -39,12 +42,24 @@ export class Message extends Base {
...
   }
+
+  public async pin(options: MessagePinOptions): Promise<Message> {
+    return new Promise((resolve, reject) => {
+      this.client.api.request({
+        method: "PUT",
+        endpoint: `/channels/channelID}/pins/${this.id}`,
+        body: options,
+      });
+
```
Comment (PraveshKunwar) [inline src/structures/Message.ts:61]:
```ts
.catch(err => {
err ? reject(this) : false
})
```

**Review by PraveshKunwar — CHANGES_REQUESTED** [review]
Look at comments made, but everything else looks good for now.

File: src/structures/Message.ts (hunk 1 above)
Comment (gautam-paranjape) [reply src/structures/Message.ts:47]: Ik, that's why I left it like that as we don't have a channel structure yet, we'll do that once we have the structure.

File: src/structures/Message.ts (hunk 2 above)
Comment (gautam-paranjape) [reply src/structures/Message.ts:61]: Got it

**Review by nerdthatnoonelikes — APPROVED** [review]
Looks good now.

**Comment by gautam-paranjape** [comment]
Wait for Pravesh to approve, then let's merge.

**Comment by PraveshKunwar** [comment]
Really good job. I will work on replacing `channelID` with `${this.channel.id}` once channel becomes a property. Looks good to merge to master.

**Review by PraveshKunwar — APPROVED** [review]
Looks good

**Pull request merged**